nut: 2.8.0 -> 2.8.2

bjornfor opened the pull request:
## Description of changes

* nut wants to build the new GPIO driver by default. Our libgpiod
  version isn't compatible and configure fails with

    checking for libgpiod version via pkg-config (1.0.0 minimum required)... 2.1.3 found
    checking for libgpiod cflags... -I/nix/store/igaf4d9zr8nmmk91icf7f2w0mw23238p-libgpiod-2.1.3/include
    checking for libgpiod ldflags... -L/nix/store/igaf4d9zr8nmmk91icf7f2w0mw23238p-libgpiod-2.1.3/lib -lgpiod
    checking for gpiod.h... yes
    checking for gpiod_chip_open_by_name... no
    checking for gpiod_chip_close... yes
    [...]
    configure: error: No supported GPIO library was found, required for GPIO driver

  So disable that for now.

* Refresh our search path patch.

* solaris-init doesn't exist in the output anymore, so we can remove the
  removal.

I tested this upgrade with my UPS on NixOS 24.05.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bjornfor committed:
nut: 2.8.0 -> 2.8.2

* nut wants to build the new GPIO driver by default. Our libgpiod
  version isn't compatible and configure fails with

    checking for libgpiod version via pkg-config (1.0.0 minimum required)... 2.1.3 found
    checking for libgpiod cflags... -I/nix/store/igaf4d9zr8nmmk91icf7f2w0mw23238p-libgpiod-2.1.3/include
    checking for libgpiod ldflags... -L/nix/store/igaf4d9zr8nmmk91icf7f2w0mw23238p-libgpiod-2.1.3/lib -lgpiod
    checking for gpiod.h... yes
    checking for gpiod_chip_open_by_name... no
    checking for gpiod_chip_close... yes
    [...]
    configure: error: No supported GPIO library was found, required for GPIO driver

  So disable that for now.

* Refresh our search path patch.

* solaris-init doesn't exist in the output anymore, so we can remove the
  removal.

JohnRTitor reviewed (dismissed):
I believe we can still build with `libgpiod_1`. https://github.com/JohnRTitor/nixpkgs/commit/24f7e2c63a6e9bfd7f0aaeb5647754a19618d4e2 (Didn't know about this PR before)

Changes mostly taken from https://github.com/NixOS/nixpkgs/issues/310868#issuecomment-2299424478

bjornfor commented:
> I believe we can still build with `libgpiod_1`. [JohnRTitor@24f7e2c](https://github.com/JohnRTitor/nixpkgs/commit/24f7e2c63a6e9bfd7f0aaeb5647754a19618d4e2) (Didn't know about this PR before)
> 
> Changes mostly taken from [#310868 (comment)](https://github.com/NixOS/nixpkgs/issues/310868#issuecomment-2299424478)

Thanks for the links. I didn't know there was an issue asking for NUT bump.

Let's merge? (We can always enable GPIO driver later if someone wants it.)

JohnRTitor commented:
I am not a fan of disabling the feature if we can enable it, especially if there is a working solution.

bjornfor committed:
nut: enable the GPIO driver

NUT wants to build it by default, so allow it. It works with libgpiod_1,
not libgpiod. The closure size increases from 365 MiB to 366 MiB.

bjornfor commented:
> I am not a fan of disabling the feature if we can enable it, especially if there is a working solution.

Ok, the GPIO driver is enabled now.

JohnRTitor commented:
@ofborg build nut

JohnRTitor reviewed (approved):
At some point, I would like to move to GitHub provided source, this is good enough for now.
